Packs/Netskope/Integrations/NetskopeEventCollector/NetskopeEventCollector.py

```diff
@@ -101,77 +101,121 @@ def populate_modeling_rule_fields(event: dict, event_type: str):
+        is_command (bool): Are we running the commands or test_module or not
+
+    Returns (list): list of all the events from a start time.
     """
     events_result = []
     if limit is None:
         limit = MAX_EVENTS_PAGE_SIZE
     for event_type in ALL_SUPPORTED_EVENT_TYPES:
-        # et - event_type
-        last_run_ids = set(last_run.get(f'{event_type}-ids', []))
         et_events: list = []
-        et_limit = limit
-        while len(et_events) < limit:
-            page_limit = min(et_limit, MAX_EVENTS_PAGE_SIZE)
-            if event_type == 'alert':
-                response = client.get_alerts_request_v1(last_run, page_limit)
-            else:
-                response = client.get_events_request_v1(event_type, last_run, page_limit)
+        next_batch = True
+        events = []
+        skip = 0
+        while next_batch:
+            if api_version == 'v1':
+                if event_type == 'alert':
+                    response = client.get_alerts_request_v1(last_run, skip, limit, is_command)
+                else:
+                    response = client.get_events_request_v1(event_type, last_run, skip, limit, is_command)
 
-            if response.get('status') != 'success' or not (results := response.get('data', [])):  # type: ignore
-                break
+                if response.get('status') != 'success':  # type: ignore
+                    break
 
-            events = []
-            for event in results:
-                if event.get('timestamp') == last_run[event_type] and event.get('_id') not in last_run_ids:
-                    events.append(event)
-                    last_run_ids.add(event['_id'])
-                else:
-                    last_run[f'{event_type}-ids'] = [event['_id']]
-                    last_run_ids = set(event['_id'])
-                    last_run[event_type] = event['timestamp']
-                    events.append(event)
-
-            et_events.extend(events)
-            # prepare for the next iteration
-            last_run[f'{event_type}-ids'] = list(last_run_ids)
-            demisto.debug(f'Initialize last run again - {last_run}')
-            et_limit -= len(events)
-            # results were less than a page - continue fetching next time
-            if len(results) < MAX_EVENTS_PAGE_SIZE:
+                results = response.get('data', [])  # type: ignore
+
+            else:  # API version == v2
+                response = client.get_events_request_v2(event_type, last_run, skip, limit, is_command)
+                if response.get('ok') != 1:
+                    break
+
+                results = response.get('result', [])
+
+            demisto.debug(f'The number of events pagination - {len(results)}')
+            events.extend(results)
+            if len(results) == MAX_EVENTS_PAGE_SIZE:
+                skip += MAX_EVENTS_PAGE_SIZE
+
+            if len(results) < MAX_EVENTS_PAGE_SIZE or not results or len(events) == MAX_SKIP:
```

Comment:
Basically, here it can't be bigger.
Since the maximum you can bring in one call is 10K, there can be only 5 calls we decided about.
It won't be bigger than the MAX_SKIP

Reply:
The user has no control on the page/batch size?

Reply:
Nope. As we specified ahead, for the fetch we will get the maximum number in one call which is 10K and we will go back in pagination only till 50K.